src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php

```diff
@@ -82,12 +80,50 @@ public function setMarking(object $subject, Marking $marking, array $context = [
             $marking = key($marking);
         }
 
-        $method = 'set'.ucfirst($this->property);
+        ($this->getSetter($subject))($marking, $context);
+    }
+
+    private function getGetter(object $subject): callable
+    {
+        $property = $this->property;
+        $method = 'get'.ucfirst($property);
+
+        return match ($this->getters[$subject::class] ??= $this->getType($subject, $property, $method)) {
+            MarkingStoreMethod::METHOD => $subject->{$method}(...),
+            MarkingStoreMethod::PROPERTY => static fn () => $subject->{$property},
+        };
+    }
+
+    private function getSetter(object $subject): callable
+    {
+        $property = $this->property;
+        $method = 'set'.ucfirst($property);
 
-        if (!method_exists($subject, $method)) {
-            throw new LogicException(sprintf('The method "%s::%s()" does not exist.', get_debug_type($subject), $method));
+        return match ($this->setters[$subject::class] ??= $this->getType($subject, $property, $method)) {
+            MarkingStoreMethod::METHOD => $subject->{$method}(...),
+            MarkingStoreMethod::PROPERTY => static fn ($marking) => $subject->{$property} = $marking,
+        };
+    }
+
+    private static function getType(object $subject, string $property, string $method): MarkingStoreMethod
+    {
+        if (method_exists($subject, $method) && (new \ReflectionMethod($subject, $method))->isPublic()) {
+            return MarkingStoreMethod::METHOD;
+        }
+
+        try {
+            if ((new \ReflectionProperty($subject, $property))->isPublic()) {
+                return MarkingStoreMethod::PROPERTY;
+            }
+        } catch (\ReflectionException) {
         }
 
-        $subject->{$method}($marking, $context);
+        throw new LogicException(sprintf('Cannot store marking: class "%s" should have either a public method named "%s()" or a public property named "$%s"; none found.', get_debug_type($subject), $method, $property));
     }
 }
+
+enum MarkingStoreMethod
```

Comment:
should be tagged as `@internal` (even if not autoloadable through PSR-4), IDEs will still discover this enum and suggest it in completion otherwise)

Reply:
Fixed, thanks